helm/cluster-azure/templates/_helpers.tpl

```diff
@@ -326,6 +326,23 @@ node-natgateway
 {{- end -}}
 {{- end -}}
 
+{{- define "network.subnet.privateEndpoints" -}}
+{{- if (gt (len $) 0) -}}
```

Comment:
```suggestion
{{- if (gt (len $.endpoints) 0) -}}
```

And just to be sure, please test that without specifying the new value, the output manifest doesn't change with this PR.